core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -234,10 +234,10 @@ abstract class RDD[T: ClassTag](
    * because DAGs are acyclic, and we only ever hold locks for one path in that DAG, there is no
    * chance of deadlock.
    *
-   * The use of Integer is simply so this is serializable -- executors may reference the shared
-   * fields (though they should never mutate them, that only happens on the driver).
+   * Executors may reference the shared fields (though they should never mutate them,
+   * that only happens on the driver).
    */
-  private val stateLock = new Integer(0)
+  private val stateLock = new Serializable {}
```

Comment:
Yeah anything Serializable is fine. Integer.valueOf(0) isn't ideal because it's going to be a shared object. Anything would work - String("") for example. This seems OK to just slightly odd looking. It could also be a proper Lock, used as a lock, but whatever.

Reply:
@srowen Do you want me to modify this?

Reply:
Seems OK to me as a minimal change that shouldn't matter for behavior.